src/shogun/regression/LeastAngleRegression.h

```diff
@@ -183,13 +205,385 @@ class CLeastAngleRegression: public CLinearMachine
 	/** @return object name */
 	virtual const char* get_name() const { return "LeastAngleRegression"; }
 
+
+	template <typename ST>
+	static vector<ST> make_vector(int32_t size, ST val)
+	{
+		vector<ST> result(size);
+		fill(result.begin(), result.end(), val);
+		return result;
+	}
+
+	template <typename ST>
+	static void plane_rot(ST x0, ST x1,
+		ST &y0, ST &y1, SGMatrix<ST> &G)
+	{
+		G.zero();
+		if (x1 == 0)
+		{
+			G(0, 0) = G(1, 1) = 1;
+			y0 = x0;
+			y1 = x1;
+		}
+		else
+		{
+			ST r = CMath::sqrt(x0*x0+x1*x1) ;
+			ST sx0 = x0 / r;
+			ST sx1 = x1 / r;
+
+			G(0,0) = sx0;
+			G(1,0) = -sx1;
+			G(0,1) = sx1;
+			G(1,1) = sx0;
+
+			y0 = r;
+			y1 = 0;
+		}
+	}
+
+	template <typename ST>
+	static void find_max_abs(const vector<ST> &vec, const vector<bool> &ignore_mask,
+		int32_t &imax, ST& vmax)
+	{
+		imax = -1;
+		vmax = -1;
+		for (uint32_t i=0; i < vec.size(); ++i)
+		{
+			if (ignore_mask[i])
+				continue;
+
+			if (CMath::abs(vec[i]) > vmax)
+			{
+				vmax = CMath::abs(vec[i]);
+				imax = i;
+			}
+		}
+	}
+
+	template <typename ST>
+	bool train_machine(CFeatures* data)
```

Comment:
I am actually not sure about the most elegant way at this moment.
We might have to do some explicit checking of types (somewhere under the hood of shogun)
@vigsterkr whats you suggestion?